src/System.Private.ServiceModel/tests/Scenarios/Client/ClientBase/DuplexClientBaseTests.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+public static class DuplexClientBaseTests
+{
+    static TimeSpan s_WaitTimeout = TimeSpan.FromSeconds(10);
+
+    [Fact]
+    [OuterLoop]
+    public static void DuplexClientBaseOfT_OverNetTcp_Call()
+    {
+        DuplexClientBase<IWcfDuplexService> duplexService = null;
+        bool success = false;
+
+        try
+        {
+            NetTcpBinding binding = new NetTcpBinding();
+            binding.Security.Mode = SecurityMode.None;
+
+            WcfDuplexServiceCallback callbackService = new WcfDuplexServiceCallback();
+            InstanceContext context = new InstanceContext(callbackService);
+
+            duplexService = new MyDuplexClientBase<IWcfDuplexService>(context, binding, new EndpointAddress(Endpoints.Tcp_NoSecurity_Callback_Address));
+            IWcfDuplexService proxy = duplexService.ChannelFactory.CreateChannel();
+
+            proxy.Ping();
+            success = callbackService.CallbackResetEvent.WaitOne(s_WaitTimeout);
+
+            duplexService.Close();
+        }
+        finally
+        {
+            if (duplexService != null && duplexService.State != CommunicationState.Closed)
+            {
+                duplexService.Abort();
+            }
+        }
+
+        Assert.True(success, "Test timed out.");
```

Comment:
I think it would be clearer to move this Assert up right after the WaitOne because "success" only means it didn't time out.  For example, if it does timeout, I'd like to see that assert fire, not the exception that might occur from the Close(). I might consider an Assert to fail if the finally requires an Abort.   That would be useful to know if that path ever executes. Not required -- your call.